src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs

```diff
@@ -7528,6 +7556,71 @@ void M()
             comp.VerifyDiagnostics();
         }
 
+        [Fact]
+        public void LambdaDefaultInDifferentNestedScope()
+        {
+            var source = """
+class Program
+{
+    void M()
+    {
+        {
+            const string s = "abcdef";
+        }
+        var _ = (string str = s) => str;
+    }
+}
+""";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (6,26): warning CS0219: The variable 's' is assigned but its value is never used
+                //             const string s = "abcdef";
+                Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "s").WithArguments("s").WithLocation(6, 26),
+                // (8,31): error CS0103: The name 's' does not exist in the current context
+                //         var _ = (string str = s) => s;
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "s").WithArguments("s").WithLocation(8, 31));
+        }
+
+        [ConditionalFact(typeof(NoIOperationValidation))]
+        public void LambdaDefaultLocalConstantSameScope_PreDefinition()
+        {
+            var source = """
+class Program
+{
+    void M()
+    {
+        const string s = "abcdef";
+        var _ = (string str = s) => s;
+    }
+}
+""";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+        }
+
+        [ConditionalFact(typeof(NoIOperationValidation))]
+        public void LambdaDefaultLocalConstantSameScope_PostDefinition()
+        {
+            var source = """
+class Program
+{
+    void M()
+    {
+        var lam = (string str = s) => { };
+        const string s = "abcdef";
+    }
+}
+""";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (5,27): error CS1750: A value of type 'var' cannot be used as a default parameter because there are no standard conversions to type 'string'
+                //         var lam = (string str = s) => { };
+                Diagnostic(ErrorCode.ERR_NoConversionForDefaultParam, "str").WithArguments("var", "string").WithLocation(5, 27),
```

Comment:
What in the world is this error?

Reply:
Good catch, I didn't notice (this was Adam's code). But it can be reproduced in `main`: [SharpLab](https://sharplab.io/#v2:EYLgtghglgdgNAFxANwKYCcoDMCeAfAAQCYACAgRgHYBYAKAG86TmyAWEgWQAoBKJlxrRbC2JADIB7AMYQANlwoAGEgGcE6EgF5VPEvRIBffiJJSJMNWXLKVWkgCIIwKQBNUWewG5jhugaA=).